pkg/declcfg/properties.go

```diff
@@ -0,0 +1,32 @@
+package declcfg
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/property"
+)
+
+func parseProperties(props []property.Property) (*property.Properties, error) {
+	out, err := property.Parse(props)
+	if err != nil {
+		return nil, err
+	}
+
+	channels := map[string]struct{}{}
+	for _, ch := range out.Channels {
+		if _, ok := channels[ch.Name]; ok {
+			return nil, propertyDuplicateError{typ: property.TypeChannel, key: ch.Name}
```

Comment:
I agree this likely doesn't belong in `pkg/declcfg`, 

However, I think this is something we would need to check when converting to the model because the GRPC API has a built-in assumption that a bundle can only replace one other bundle in a given channel, and the model currently encodes that assumption in its `Bundle` type, which has fields for `Channel` and `Replaces`.

Reply:
I just checked. This code only runs when converting to the model, not during `declcfg.LoadDir`. The only validations that occur at parse time are:
1. If it isn't JSON, ignore it.
1. If it doesn't parse as a `Meta`, ignore it.
1. If it has an empty "schema" field, ignore it.
1. If "schema" == "olm.package", does it the object unmarshal into a `Package` object successfully?
1. If "schema" == "olm.bundle", does it the object unmarshal into a `Bundle` object successfully?
1. If "schema" is anything else, just dump it into the `Others` bucket.